src/NuGet.Core/NuGet.Protocol/Exceptions/FatalProtocolException.cs

```diff
@@ -15,5 +15,17 @@ public FatalProtocolException(string message) : base(message)
         public FatalProtocolException(string message, Exception innerException) : base(message, innerException)
         {
         }
+
+        public FatalProtocolException(string message, NuGetLogCode logCode) : base(message)
+        {
+            LogCode = logCode;
+        }
+
+        public FatalProtocolException(string message, Exception innerException, NuGetLogCode logCode) : base(message, innerException)
+        {
+            LogCode = logCode;
+        }
+
+        public NuGetLogCode LogCode { private set; get; }
```

Comment:
1. the private setter allows any method in the class to change the value, but if you omit the setter at all (so it's a get only property), then the value can be set only in a constructor, as it is in this class.

2. idealistic nitpick:  From an API design point of view, NuGetLogCode just doesn't feel right to me. The name suggests that it's the codes used when logging, but this is an API for software developers to use one specific assembly from the suite that represent NuGet from a user-experience.  My first reaction is that an `int` representing the HTTP status would be better, but `NuGet.Protocol` is supposed to be an abstraction of wheverever packages are coming from, so using HTTP status would be a leaky abstraction. I don't think anyone would support a suggestion to create a FatalProtocolExceptionReason enum, so I guess the type is fine as it is. I just feel like there have been times in the past when we re-used existing types and that led to problems later because the context the type was being used in was different, and modifying the type for one use-case doesn't make sense, or causes bugs, in the other use cases.

3. The two existing constructors that don't define a value for `LogCode` look like a bug. You should have those constuctors call the overloads that accept a `logCode` parameter, with a default value.

Reply:
thanks. i will consider all that and also look at the other exception in our codebase that does something like this now... to see if it uses logcode or not.

Reply:
i went with HttpStatusCode enum... Let me know what you think.

i looked at PackagingException...and it didn't have it on the exception...it had it on PackagingLogMessage ... which wouldn't help us...cause the exception needs to carry the data.